File: Ejercicios/Calculator/Net/Solucion/Calculator.Specs/Steps/AdditionSteps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Calculator.Library;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using TechTalk.SpecFlow;

namespace Calculator.Specs.Steps
{
    [Binding]
    public class AdditionSteps
    {
        Calc calculator=new Calc();
        private int resultCalculator = 0;

        [Given(@"I have entered (.*) into the calculator")]
        public void GivenIHaveEnteredIntoTheCalculator(int number)
        {
            calculator.Enter(number);
        }
        
        [When(@"I press add")]
        public void WhenIPressAdd()
        {
            resultCalculator = calculator.Add();
        }
        
        [Then(@"the result should be (.*) on the screen")]
        public void ThenTheResultShouldBeOnTheScreen(int result)
        {
            Assert.AreEqual(result, resultCalculator);
        }

    }
}

File: Ejercicios/Calculator/Java/Solucion/src/test/java/steps/AdditionSteps.java
package steps;

import static org.junit.Assert.*;

import org.jbehave.core.annotations.Given;
import org.jbehave.core.annotations.Pending;
import org.jbehave.core.annotations.Then;
import org.jbehave.core.annotations.When;

import domain.Calculator;

public class AdditionSteps {

	Calculator calculator = new Calculator();
	int resultCalculator = 0;

	@Given("I have entered $number into the calculator")
	public void givenIHaveEnteredIntoTheCalculator(int number) {
		calculator.Enter(number);
	}

	@When("I press add")
	public void whenIPressAdd() {
		resultCalculator = calculator.Add();
	}

	@Then("the result should be 120 on the screen")
	public void thenTheResultShouldBe120OnTheScreen() {
		assertEquals(120, resultCalculator);
	}
}
